Source/Krypton Components/Krypton.Ribbon/Controls Visuals/VisualPopupAppMenu.cs

```diff
@@ -512,50 +515,48 @@ private void SetPalette(PaletteBase? palette)
                 _palette = palette;
 
                 // Update redirector to use palette as source for obtaining values
-                Redirector.Target = _palette;
+                if (Redirector != null) Redirector.Target = _palette;
 
                 // Get the renderer associated with the palette
-                Renderer = _palette.GetRenderer();
-
-                // Hook to new palette events
                 if (_palette != null)
                 {
-                    _palette.PalettePaint += OnPaletteNeedPaint;
-                    _palette.BasePaletteChanged += OnBaseChanged;
-                    _palette.BaseRendererChanged += OnBaseChanged;
+                    Renderer = _palette.GetRenderer();
+
+                    // Hook to new palette events
+                    if (_palette != null)
+                    {
+                        _palette.PalettePaint += OnPaletteNeedPaint;
+                        _palette.BasePaletteChanged += OnBaseChanged;
+                        _palette.BaseRendererChanged += OnBaseChanged;
+                    }
                 }
             }
         }
 
         private void OnBaseChanged(object sender, EventArgs e)
         {
             // Change in base renderer or base palette require we fetch the latest renderer
-            Renderer = _palette.GetRenderer();
+            if (_palette != null) Renderer = _palette.GetRenderer();
```

Comment:
brackets for all `if`'s please

Reply:
All `if` statements in entire solution now have brackets